File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Querying/HowToStreamQueryResults.cs
using System.Collections.Generic;

using Raven.Abstractions.Data;
using Raven.Client.Document;
using Raven.Json.Linq;

namespace Raven.Documentation.Samples.ClientApi.Commands.Querying
{
	public class HowToStreamQueryResults
	{
		private interface IFoo
		{
			#region stream_query_1
			IEnumerator<RavenJObject> StreamQuery(
				string index,
				IndexQuery query,
				out QueryHeaderInformation queryHeaderInfo);
			#endregion
		}

		public HowToStreamQueryResults()
		{
			using (var store = new DocumentStore())
			{
				#region stream_query_2
				QueryHeaderInformation queryHeaderInfo;
				var enumerator = store
					.DatabaseCommands
					.StreamQuery(
						"Orders/Totals",
						new IndexQuery
						{
							Query = "Company:companies/1"
						},
						out queryHeaderInfo);

				while (enumerator.MoveNext())
				{
					var order = enumerator.Current;
				}
				#endregion
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/querying/HowToStreamQueryResults.java
package net.ravendb.clientapi.commands.querying;

import net.ravendb.abstractions.basic.CloseableIterator;
import net.ravendb.abstractions.basic.Reference;
import net.ravendb.abstractions.data.IndexQuery;
import net.ravendb.abstractions.data.QueryHeaderInformation;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;


public class HowToStreamQueryResults {

  @SuppressWarnings("unused")
  private interface IFoo {

    //region stream_query_1
    public CloseableIterator<RavenJObject> streamQuery(String index, IndexQuery query, Reference<QueryHeaderInformation> queryHeaderInfo);
    //endregion
  }

  @SuppressWarnings("unused")
  public HowToStreamQueryResults() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region stream_query_2
      Reference<QueryHeaderInformation> queryHeaderInfoRef = new Reference<>();
      try (CloseableIterator<RavenJObject> iterator = store.getDatabaseCommands().streamQuery("Orders/Totals", new IndexQuery("Company:companies/1"), queryHeaderInfoRef)) {

        while (iterator.hasNext()) {
          RavenJObject order = iterator.next();
        }
      }


      //endregion
    }
  }
}
